mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -212,12 +230,139 @@ class LinearRegressionModel private[ml] (
   extends RegressionModel[Vector, LinearRegressionModel]
   with LinearRegressionParams {
 
+  @transient private var trainingSummary: Option[LinearRegressionTrainingSummary] = None
```

Comment:
@dbtsai I would prefer to have stats included within the same object, rather than having separate objects for small (local) stats vs. large (distributed) stats.  Do you have a good use case for when a worker would need to access these stats?  Even in that case, the driver could get the needed stats and then pass those to the worker.